src/main/java/org/jbake/app/ContentStore.java

```diff
@@ -0,0 +1,167 @@
+import com.orientechnologies.orient.core.sql.query.OSQLSynchQuery;
+import java.util.List;
+import static org.jbake.app.DBUtil.updateSchema;
+import org.jbake.model.DocumentTypes;
+
+/**
+ *
+ * @author jdlee
+ */
+public class ContentStore {
+    private ODatabaseDocumentTx db;
+    
+    public ContentStore(final String type, String name) {
+        db = new ODatabaseDocumentTx(type + ":" + name);
+        boolean exists = db.exists();
+        if (!exists) {
+            db.create();
+        }
+        db = ODatabaseDocumentPool.global().acquire(type + ":" + name, "admin", "admin");
+        ODatabaseRecordThreadLocal.INSTANCE.set(db);
+        if (!exists) {
+            updateSchema();
+        }
+    }
+    
+    public  void updateSchema() {
+        OSchema schema = db.getMetadata().getSchema();
+        for (String docType : DocumentTypes.getDocumentTypes()) {
+            if (schema.getClass(docType)==null) {
+                createDocType(schema, docType);
+            }
+        }
+        if (schema.getClass("Signatures")==null) {
+            // create the sha1 signatures class
+            OClass signatures = schema.createClass("Signatures");
+            signatures.createProperty("key", OType.STRING).setNotNull(true);
+            signatures.createProperty("sha1", OType.STRING).setNotNull(true);
+        }
+    }
+
+
+
+    public void close() {
+        db.close();
+    }
+
+    public void drop() {
+        db.drop();
+    }
+
+    public long countClass(String iClassName) {
+        return db.countClass(iClassName);
+    }
+
+    private static void createDocType(final OSchema schema, final String doctype) {
+        OClass page = schema.createClass(doctype);
+        page.createProperty("sha1", OType.STRING).setNotNull(true);
+        page.createProperty("sourceuri", OType.STRING).setNotNull(true);
+        page.createProperty("rendered", OType.BOOLEAN).setNotNull(true);
+        page.createProperty("cached", OType.BOOLEAN).setNotNull(true);
+
+        // commented out because for some reason index seems to be written
+        // after the database is closed to this triggers an exception
+        //page.createIndex("uriIdx", OClass.INDEX_TYPE.UNIQUE, "uri");
+        //page.createIndex("renderedIdx", OClass.INDEX_TYPE.NOTUNIQUE, "rendered");
+    }    
+
+    public List<ODocument> getPublishedPosts() {
+        return getPublishedContent("post");
+    }
+
+    public List<ODocument> getPublishedPages() {
+        return getPublishedContent("page");
+    }
+
+    public List<ODocument> getPublishedContent(String docType) {
+        return query("select * from "+docType+" where status='published' order by date desc");
+    }
+
+    public List<ODocument> getAllContent(String docType) {
+        return query("select * from "+docType+" order by date desc");
+    }
+
+    public List<ODocument> getAllTags() {
+        return query("select tags from post where status='published'");
+    }
```

Comment:
That method name should reflect the fact that the tags are extracted from published posts.